play.pokemonshowdown.com/src/panel-chat.tsx

```diff
@@ -176,16 +177,43 @@ export class ChatRoom extends PSRoom {
 			this.title = `[DM] ${nameWithGroup.trim()}`;
 		}
 	}
-	handleHighlight = (message: string, name: string) => {
-		if (!PS.prefs.noselfhighlight && PS.user.nameRegExp?.test(message)) {
-			this.notify({
+	handleHighlight = (args: Args) => {
+		let name;
+		let message;
+		let msgTime = 0;
+		if (args[0] === 'c:') {
+			msgTime = parseInt(args[1]);
+			name = args[2];
+			message = args[3];
+		} else {
+			name = args[1];
+			message = args[2];
+		}
+		let lastMessageDates = Dex.prefs('logtimes') || (PS.prefs.set('logtimes', {}), Dex.prefs('logtimes'));
+		if (!lastMessageDates[PS.server.id]) lastMessageDates[PS.server.id] = {};
+		let lastMessageDate = lastMessageDates[PS.server.id][this.id] || 0;
+		// because the time offset to the server can vary slightly, subtract it to not have it affect comparisons between dates
+		let serverMsgTime = msgTime - (this.timeOffset || 0);
+		let mayNotify = serverMsgTime > lastMessageDate && name !== PS.user.userid;
+		if (PS.isVisible(this)) {
+			this.lastMessageTime = null;
+			lastMessageDates[PS.server.id][this.id] = serverMsgTime;
+			PS.prefs.set('logtimes', lastMessageDates);
+		} else {
+			// To be saved on focus
+			let lastMessageTime = this.lastMessageTime || 0;
+			if (lastMessageTime < serverMsgTime) this.lastMessageTime = serverMsgTime;
+		}
+		if (PS.getHighlight(message, this.id)) {
+			if (mayNotify) this.notify({
 				title: `Mentioned by ${name} in ${this.id}`,
 				body: `"${message}"`,
 				id: 'highlight',
 			});
 			return true;
 		}
 		/*
+		// i moved the following logic to ``PS.getHighlight``
```

Comment:
You can just remove this (and the comment after it, no need to keep it around after this.